File: src/main/java/no/jstien/bikeapi/utils/holiday/HolidayRegistry.java
package no.jstien.bikeapi.utils.holiday;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.util.Calendar;
import java.util.List;

public class HolidayRegistry {
    private static final Logger LOG = LogManager.getLogger();

    private HolidayRepository holidayRepository;

    public HolidayRegistry(HolidayRepository holidayRepository) {
        this.holidayRepository = holidayRepository;
    }

    public boolean isHoliday(Calendar date) {
        List<Holiday> holidays = holidayRepository.getHolidaysForCurrentYear();
        boolean result = holidays.stream()
                .filter(h -> h.getDate().equals(date))
                .count() > 0;
        return result;
    }

}

File: src/main/java/no/jstien/bikeapi/utils/holiday/HolidayRegistry.kt
package no.jstien.bikeapi.utils.holiday

import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger

import java.util.Calendar

open class HolidayRegistry(private val holidayRepository: HolidayRepository) {

    open fun isHoliday(date: Calendar): Boolean {
        val holidays = holidayRepository.holidaysForCurrentYear
        return holidays.stream()
                .filter { h -> h.date == date }
                .count() > 0
    }

    companion object {
        private val LOG = LogManager.getLogger()
    }

}
